dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/RelationshipsValidationHook.java

```diff
@@ -130,128 +148,58 @@ private void validateAutoRelationship( ValidationErrorReporter reporter, Relatio
         }
     }
 
-    private void validateRelationshipType( ValidationErrorReporter reporter,
-        Relationship relationship, TrackerPreheat preheat )
+    private String relationshipItemValueType( RelationshipItem item )
     {
-        List<RelationshipType> relationshipsTypes = preheat.getAll( TrackerIdScheme.UID, RelationshipType.class );
-        Optional<RelationshipType> optionalRelationshipType = relationshipsTypes.stream()
-            .filter( type -> type.getUid().equals( relationship.getRelationshipType() ) ).findFirst();
-
-        if ( !optionalRelationshipType.isPresent() )
-        {
-            addError( reporter, E4004, RELATIONSHIP_TYPE );
-            return;
-        }
-
-        if ( relationship.getFrom() == null )
+        if ( StringUtils.isNotEmpty( item.getTrackedEntity() ) )
         {
-            addError( reporter, E4004, "relationship.getFrom" );
+            return TRACKED_ENTITY;
         }
-        else
-        {
-            optionalRelationshipType
-                .ifPresent( relationshipType -> validateRelationshipConstraint( relationship.getFrom(),
-                    relationshipType.getFromConstraint() )
-                        .stream()
-                        .forEach( reporter::addError ) );
-        }
-
-        if ( relationship.getTo() == null )
+        else if ( StringUtils.isNotEmpty( item.getEnrollment() ) )
         {
-            addError( reporter, E4004, "relationship.getTo" );
+            return ENROLLMENT;
         }
-        else
+        else if ( StringUtils.isNotEmpty( item.getEvent() ) )
         {
-            if ( optionalRelationshipType.isPresent() )
-            {
-                validateRelationshipConstraint( relationship.getTo(),
-                    optionalRelationshipType.get().getToConstraint() )
-                    .stream()
-                    .forEach( reporter::addError );
-                ;
-            }
+            return EVENT;
         }
-
+        return null;
     }
 
-    private List<TrackerErrorReport.TrackerErrorReportBuilder> validateRelationshipConstraint( RelationshipItem item,
-        RelationshipConstraint relationshipType )
+    private List<TrackerErrorReport.TrackerErrorReportBuilder> validateRelationshipConstraint( String relSide,
```

Comment:
This is just a note, more than feedback:

I was thinking we have to validate that wrong values are not inserted as well (IE. tei inserted, when constraint is EVENT).

However, we actually ignore other values when comitting, so I say we don't add that validation, unless you think it makes sense to report sending too much data (wrong data is already covered, assuming wrong data = wrong constraint type)

Reply:
@stian-sandvold Ah, I didn't know. I'll ping you so we can discuss.